src/matchers.rs

```diff
@@ -7,24 +7,27 @@
 use serde_json::Value;
 use std::iter;
 
+/// An iterator over matcher selection results.
+type Iter<'a> = Box<dyn Iterator<Item = &'a Value> + 'a>;
```

Comment:
Hmmm. I wonder if `Iter` could be confusing. How about `SelectionIter`?

Reply:
I took inspiration from https://doc.rust-lang.org/std/option/struct.Iter.html
and thought perhaps Rust also had the "no stutter rule" (which states of a module name is foo, don't call types "things unnecessarily foo::Foo), but I guess Rust is a bit different since you often `use` unqualified identifiers, so std::option::Option does exist (but perhaps one is not supposed to `use std::option::Iter` since it's an implementation detail?)

Reply:
Ok, let's stick with `Iter` and see how it pans out.